### ISSUE TITLE
Build System Issues

### ISSUE THREAD

**Issue by BrodyEller** [kidrigger/godot-videodecoder#22]
I know I messaged in a now closed issue about a month and a half ago about the build setup of this repo. Since then, major changes have been merged that significantly change the way everything is built. While I agreed with @jamie-pate that using the system installed ffmpeg libs wasn't great, I disagree with the build setup as it stands right now.

As far as I can tell, support for compiling on macOS and Windows has been dropped in favor of cross-compiling on Linux. It's nice to be able to cross-compile but I do a significant portion of development on macOS and I feel compiling from the host system to the host system should be the priority.

As for how to manage the ffmpeg dependency, I'm proposing that the user uses their system package manager(pacman, brew, MSYS2) to install ffmpeg dependencies, e.g. libvorbis, x264, etc. Then a specific release/commit of ffmpeg is cloned to the user's machine, compiled with the specified flags, and installed to a third party directory in the godot-videodecoder folder. Then we can point scons to this directory.

I know you guys have been up to a lot recently in changing the build system but at the moment it's broken for me being on macOS. I've been looking through jamie's repos and it kind of looks like he's trying to set up docker? I'm not sure, and I'm definitely no expert on build systems but let me know what you think!

**Comment by jamie-pate** [comment]
https://github.com/kidrigger/godot-videodecoder/pull/23 was merged about an hour after this comment :)

Not sure how compatible Linux Docker images are with mac, but it should be a start.

My goal is to make it simple to build portable libs that can be bundles with a release for the major platforms. I have a very tight deadline, but spent the extra bit of time to get all three working in Docker because it's really really nice for repeatability.

I tried to preserve the scons builds for OSX, but haven't tested there. If the Docker container works then I'd suggest that it's not important to have native builds (but a pr is welcome :) )
